### PR TITLE
Improve structure and add more examples

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description
Give the hackathonStarter a tune up, organize the examples, add more examples, improve testing, and demonstrate the complete task lifecycle from `createTask` to `claimPayout`.

## TODO

- [X] The examples are broken up into smaller parts and live in an `examples` directory.

- [X] The root example in `index.js` supports multiple connections to the network using a state object for each account that stores the network client and colony client for the connected account (this allows us to demonstrate the complete task lifecycle without having to reconnect to the network each time a different account needs to interact with the network).

- [X] The root example in `index.js` provides a means for storing incomplete multisig operations using an object that is accessible within each example (this allows us to restore the operation from another account in order to demonstrate the complete task lifecycle).

- [X] More comments have been added to make it more clear to developers why we are doing what.

- [X] More logs have been added and the logs are easy to read when the root example is executed.

- [X] Additional examples have been added to demonstrate customizations and funding a colony (ie, add global skill, add domain, mint tokens, claim funds, move funds between pots)

- [X] Additional examples have been added to demonstrate the complete task lifecycle (ie, update task, sign task, submit work, rate worker and manager, reveal ratings, finalize task, claim payouts)

- [X] The README has been updated

## Dependencies

- `"bn.js": "^4.11.8"`

### CODE REVIEW

File: src/examples/signTaskDueDate.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  if (operationJSON) {
+
+    // Restore operation
+    const operation = await colonyClient.setTaskDueDate.restoreOperation(
+      operationJSON,
+    );
+
+    // Check if the operation matches the colony contract and the task id and
+    // check if current account is required to sign the operation.
+    if (
```
Comment (JamesLefrere) [inline src/examples/signTaskDueDate.js]: Should we perhaps log something out if this check doesn't pass?

File: src/examples/signTaskEvaluatorPayout.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  if (operationJSON) {
+
+    // Restore operation
+    const operation = await colonyClient.setTaskEvaluatorPayout.restoreOperation(
+      operationJSON,
+    );
+
+    // Check if the operation matches the colony contract and the task id and
+    // check if current account is required to sign the operation.
+    if (
```
Comment (JamesLefrere) [inline src/examples/signTaskEvaluatorPayout.js]: As above, we should probably log something out if this doesn't work.

File: src/examples/signTaskWorkerPayout.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      operationJSON,
+    );
+
+    // Check if the operation matches the colony contract and the task id and
+    // check if current account is required to sign the operation.
+    if (
+      operation.payload.sourceAddress === colonyClient._contract.address &&
+      operation.payload.inputValues.taskId === taskId &&
+      operation.requiredSignees.includes(colonyClient.adapter.wallet.address)
+    ) {
```
Comment (JamesLefrere) [inline src/examples/signTaskWorkerPayout.js]: ☝️

File: src/index.js [hunk 4]
```diff
@@ -1,8 +1,463 @@
...
+  state.networkClient[0] = await connectNetwork(
+    0,                              // accountIndex
+  );
+
+  console.log('\n\x1b[32m' + 'account[0] createToken:' + '\x1b[0m\n');
+
+  // Create a new ERC20 token using the "createToken" example and then store
+  // the returned "tokenAddress" in the state object.
+  state.tokenAddress = await createToken(
+    state.networkClient[0],         // networkClient
```
Comment (JamesLefrere) [inline src/index.js]: A disadvantage of using these functions with arguments like this is that comments need to be added to each argument for clarity here (as opposed to using an object). I'm ok with it, so long as it's maintained as changes are made to the contracts/library.

File: src/examples/signTaskDueDate.js (hunk 1 above)
Comment (ryanchristo) [reply src/examples/signTaskDueDate.js]: :+1:

File: src/examples/signTaskEvaluatorPayout.js (hunk 2 above)
Comment (ryanchristo) [reply src/examples/signTaskEvaluatorPayout.js]: :+1:

File: src/examples/signTaskWorkerPayout.js (hunk 3 above)
Comment (ryanchristo) [reply src/examples/signTaskWorkerPayout.js]: :+1:

File: src/index.js (hunk 4 above)
Comment (ryanchristo) [reply src/index.js]: Yup, I agree. I do like that it helps differentiate the examples from the actual colonyJS functions. I think I will leave it as is for now and revisit later. Thanks!

**Pull request merged**